emgapi/management/commands/clean_empty_studies.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2017-2022 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+
+from django.db.models import Count
+from django.core.management import BaseCommand
+from django.conf import settings
+
+from emgapi.models import Study, StudySample, AnalysisJob, Run, Status, Sample
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Check for missing download files for all (or any) study."
+
+    def handle(self, *args, **kwargs):
+        logger.info("Starting to cleanup the 'empty studies'")
+
+        studies = Study.objects.exclude(is_public=Status.SUPRESSED)
+        for study in studies:
+            # only those with no jobs
+            if AnalysisJob.objects.filter(study=study).exists():
+                continue
```

Comment:
Can this be done with one query with an annotate? Not sure... If it isn't painfully slow then no worries anyway.

Reply:
It's fast enough..